File: 58-length-of-last-word/length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int i = s.length() - 1;
        while (i >= 0 && s.charAt(i) == ' ') {
            i = i - 1;
        }
        int res = 0;
        while (i >= 0 && s.charAt(i) != ' ') {
            res = res + 1;
            i -= 1;
        }
        return res;
    }
}

File: 58-length-of-last-word/length-of-last-word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """:type s: str :rtype: int"""
        # Start from the end of the string
        i = len(s) - 1
        
        # Skip trailing spaces
        while i >= 0 and s[i] == ' ':
            i -= 1
            
        # Count the length of the last word
        length = 0
        while i >= 0 and s[i] != ' ':
            length += 1
            i -= 1
            
        return length
